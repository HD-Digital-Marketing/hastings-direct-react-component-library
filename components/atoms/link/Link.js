import React from 'react'
import classNames from 'classnames'
import PropTypes from 'prop-types'
import './link.scss'

const Link = props => {

	const {
		href = '',
		rel = '',
		role = '',
		target = '',
		theme = '',
		children
	} = props

	const linkClass = classNames('a-link', className, {
		'a-link--primary': theme === 'primary',
		'a-link--secondary': theme === 'secondary'
	})

	return(
		<a {...{ href, rel, role, target }} className={linkClass}>{children}</a>
	)
}

Link.propTypes = {
	href: PropTypes.string,
	rel: PropTypes.string,
	role: PropTypes.string,
	target: PropTypes.string,
	theme: PropTypes.string,
	children: PropTypes.string
}

export default Link
