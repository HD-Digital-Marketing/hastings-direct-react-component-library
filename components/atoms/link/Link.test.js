import React from 'react';
import Link from './Link';

describe('Check the link renders', () => {
	test('it should render the link', () => {

		const renderedLink = Link({
			href: '#',
			rel: 'noopener noreferrer',
			role: 'button',
			target: '_blank',
			theme: 'primary',
			children: 'Click me'
		});

		expect(renderedLink).toEqual(<a href="#" className="a-link a-link--primary" rel="noopener noreferrer" target="_blank" role="button">Click me</a>);
	});
});