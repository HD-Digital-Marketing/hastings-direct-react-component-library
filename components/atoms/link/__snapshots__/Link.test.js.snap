// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check the link renders it should match the snapshot 1`] = `
<a
  className="a-link a-link--primary"
  href="https://google.com"
  rel="noopener noreferrer"
  role="button"
  target="_blank"
>
  Click me I'm a link
</a>
`;
